src/mirakuru/base.py

```diff
@@ -43,53 +49,94 @@
+        except psutil.NoSuchProcess:
+            pass
+        else:
+            penv = pinfo.get('environ')
+            if penv and penv.get(env_name) == env_value:
+                pids.add(pinfo['pid'])
 
-def processes_with_env(env_name, env_value):
+    return pids
+
+
+def processes_with_env_ps(env_name, env_value):
     """
     Find PIDs of processes having environment variable matching given one.
 
-    Function uses `$ ps e -ww` command so it works only on systems
+    It uses `$ ps xe -o pid,cmd` command so it works only on systems
     having such command available (Linux, MacOS). If not available function
     will just log error.
 
     :param str env_name: name of environment variable to be found
     :param str env_value: environment variable value
-    :return: process ids (PIDs) of processes that have certain environment
-             variable with certain value
+    :return: process identifiers (PIDs) of processes that have certain
+             environment variable equal certain value
     :rtype: set
     """
     pids = set()
-
+    ps_xe = ''
     try:
-        ps_xe = subprocess.check_output(('ps', 'xe', '-ww')).splitlines()
+        cmd = 'ps', 'xe', '-o', 'pid,cmd'
+        ps_xe = subprocess.check_output(cmd).splitlines()
     except OSError as err:
         if err.errno == 2:
-            log.error("`$ ps xe -ww` command was called but it is not "
+            log.error("`$ ps xe -o pid,cmd` command was called but it is not "
                       "available on this operating system. Mirakuru will not "
                       "be able to list the process tree and find if there are "
                       "any leftovers of the Executor.")
-    else:
-        env = '{0}={1}'.format(env_name, env_value)
+            return pids
+    except subprocess.CalledProcessError:
+        log.error("`$ ps xe -o pid,cmd` command exited with non-zero code.")
 
-        for line in ps_xe:
-            line = str(line)
-            if env in line:
-                pids.add(int(PS_XE_PID_MATCH.match(line).group(1)))
+    env = '{0}={1}'.format(env_name, env_value)
+
+    for line in ps_xe:
+        line = str(line)
+        if env in line:
+            pids.add(int(PS_XE_PID_MATCH.match(line).group(1)))
     return pids
 
 
+if psutil_importable is True:
+    processes_with_env = processes_with_env_psutil
+else:
+    # In case psutil can't be imported (on pypy3) we try to use '$ ps xe'
+    processes_with_env = processes_with_env_ps
+
+
 @atexit.register
 def cleanup_subprocesses():
     """On python exit: find possibly running subprocesses and kill them."""
     pids = processes_with_env(ENV_UUID, os.getpid())
     for pid in pids:
         log.debug("Killing process %d ...", pid)
-        os.kill(pid, signal.SIGKILL)
-        log.debug("Killed process %d.", pid)
+        try:
+            os.kill(pid, signal.SIGKILL)
+        except OSError as err:
+            log.warning("Could not kill the %d leaked process: %r", pid, err)
```

Comment:
I'm not sure if we should warn if the error occurs due to `ESRCH`, e.g. when the process exited on its own before we killed it.